### PR TITLE
To Do list: add & remove

### PR DESCRIPTION
**Pull request by jssol**
## Technical requirements 🎯 

- [x] Remove all hardcoded items from the tasks array.
- [x] Create a new JavaScript file for the new functionality.
- [x] Implement a function for adding a new task (add a new element to the array).
- [x] Implement a function for deleting a task (remove an element from the array).
- [x] Implement a function for editing task descriptions.
- [x] By default, new tasks should have the property completed set to false and the property index set to the value of the new array length.
- [x] Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- [x] All changes to the To-Do List should be saved in local storage.

## Useful links

- [Full project requirements](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)
- [Live Demo](https://jssol.github.io/todo-list/dist/)

### CODE REVIEW

File: src/taskManager.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  }
+
+  addTask = (description) => {
+    const task = new Task(description, this.tasks);
+    this.tasks.push(task);
+  }
+
+  // updateTask = () => {
+
+  // }
```
Comment (julie-ify) [inline src/taskManager.js]: - [x] Seems this commented code is unnecessary. Kindly remove it in order to achieve clean code.

File: src/taskManager.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  clearAll = () => {
+    this.tasks = [];
+  }
+
+  // clearComplete = () => {
+
+  // }
+}
```
Comment (julie-ify) [inline src/taskManager.js]: - [x] Seems this commented code is unnecessary. Kindly remove it in order to achieve clean code.

File: src/taskManager.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        if (li.classList.contains('completed')) {
+          check.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2m-6 9l2 2 4-4" />
+          </svg>`;
+        } else {
+          check.innerHTML = `<svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor">
+          <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M9 5H7a2 2 0 00-2 2v12a2 2 0 002 2h10a2 2 0 002-2V7a2 2 0 00-2-2h-2M9 5a2 2 0 002 2h2a2 2 0 002-2M9 5a2 2 0 012-2h2a2 2 0 012 2" />
+          </svg>`;
+        }
+      });
```
Comment (julie-ify) [inline src/taskManager.js]: - [x] Currently, the local storage does not get updated when the checkbox is toggled. The required behavior is that the local storage gets updated and toggles completed from `false` to `true`. Please kindly fix this by updating the local storage when there is a toggle of the checkbox.

chrome_gywwFxKx8O

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @jssol,

Good job so far!

## Highlight!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

- [x] Why don't you add instructions on how to run your project locally. Please include commands like `npm install` and `npm start` so that users can see how to run your app successfully locally.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elyor-doniyorov — APPROVED** [review]
Hi @jssol, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**